### PR TITLE
batchExecute functionality for SQL connector

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
Implement batchExecute functionality for SQL connector.
Partially fix https://github.com/ballerina-platform/ballerina-lang/issues/23565

## Approach
Used JDBC PreparedStatement batchExecute()

## Samples
```ballerina
import ballerina/io;
import ballerina/mysql;
import ballerina/sql;

public function main() {

    sql:Error? err = initializeClients();
    if (err is sql:Error) {
        io:println("Error occured", err);
    } else {
        io:println("Sample executed successfully!");
    }
}

function initializeClients(boolean test = false) returns sql:Error? {

    mysql:Client mysqlClient = check new ("localhost", "test", "", "test");
    io:println("MySQL client with user and password created.");

    _ = check mysqlClient->execute("DROP TABLE Persons");
    _ = check mysqlClient->execute("CREATE TABLE Persons (id int(11) NOT NULL AUTO_INCREMENT, name varchar(255) DEFAULT NULL, PRIMARY KEY (`id`))");

     sql:ParameterizedString sqlQuery1 = {
         parts : ["INSERT INTO Persons (name) values (",")"],
         insertions: ["Alice"]
     };

     sql:ParameterizedString sqlQuery2 = {
         parts : ["INSERT INTO Persons (name) values (",")"],
         insertions: ["Alice"]
     };

     sql:ParameterizedString sqlQuery3 = {
         parts : ["INSERT INTO Persons (name) values (",")"],
         insertions: ["Alice"]
     };

     sql:ParameterizedString[] sqlQueriues = [sqlQuery1, sqlQuery2, sqlQuery3];

    sql:ExecuteResult[]? result = check mysqlClient->batchExecute(sqlQueries);
    io:println(result);
    check mysqlClient.close();
}
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [x] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [x] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [x] Ballerina By Examples

### CODE REVIEW

File: stdlib/database/jdbc/src/test/java/org/ballerinalang/jdbc/batchexecute/BatchExecuteTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+import java.io.File;
+import java.sql.SQLException;
+
+import static org.ballerinalang.sql.Constants.ErrorRecordFields.EXECUTION_RESULTS;
+
+/**
+ * This test class verifies the behaviour of the batch execute.
+ *
+ * @since 1.3.0
```
Comment (sinthuja) [inline stdlib/database/jdbc/src/test/java/org/ballerinalang/jdbc/batchexecute/BatchExecuteTest.java:43]: Isn't this should be 2.0.0?

File: stdlib/database/jdbc/src/test/java/org/ballerinalang/jdbc/batchexecute/BatchExecuteTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+public class BatchExecuteTest {
+    private CompileResult result;
+    private static final String DB_NAME = "TEST_SQL_BATCH_EXECUTE";
+    private static final String JDBC_URL = "jdbc:h2:file:" + SQLDBUtils.DB_DIR + DB_NAME;
+    private BValue[] args = {new BString(JDBC_URL), new BString(SQLDBUtils.DB_USER),
+            new BString(SQLDBUtils.DB_PASSWORD)};
+    @BeforeClass
+    public void setup() throws SQLException {
+        result = BCompileUtil.compileOffline(SQLDBUtils.getBalFilesDir("batchexecute",
+                                                                            "batch-execute-query-test.bal"));
```
Comment (sinthuja) [inline stdlib/database/jdbc/src/test/java/org/ballerinalang/jdbc/batchexecute/BatchExecuteTest.java:54]: Looks like a formatting issue?

Comment (niveathika) [reply stdlib/database/jdbc/src/test/java/org/ballerinalang/jdbc/batchexecute/BatchExecuteTest.java]: Fixed in e00abad

File: stdlib/database/jdbc/src/test/java/org/ballerinalang/jdbc/batchexecute/BatchExecuteTest.java (hunk 1 above)
Comment (niveathika) [reply stdlib/database/jdbc/src/test/java/org/ballerinalang/jdbc/batchexecute/BatchExecuteTest.java]: Fixed in e00abad

**Pull request merged**